pkg/registry/storage/csidriver/strategy_test.go

```diff
@@ -74,6 +78,169 @@ func TestCSIDriverStrategy(t *testing.T) {
+		APIGroup:   "storage.k8s.io",
+		APIVersion: "v1beta1",
+		Resource:   "csidrivers",
+	})
+
+	attachRequired := true
+	podInfoOnMount := true
+	driverWithoutModes := &storage.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+		},
+		Spec: storage.CSIDriverSpec{
+			AttachRequired: &attachRequired,
+			PodInfoOnMount: &podInfoOnMount,
+		},
+	}
+	driverWithPersistent := &storage.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+		},
+		Spec: storage.CSIDriverSpec{
+			AttachRequired: &attachRequired,
+			PodInfoOnMount: &podInfoOnMount,
+			VolumeLifecycleModes: []storage.VolumeLifecycleMode{
+				storage.VolumeLifecyclePersistent,
+			},
+		},
+	}
+	driverWithEphemeral := &storage.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+		},
+		Spec: storage.CSIDriverSpec{
+			AttachRequired: &attachRequired,
+			PodInfoOnMount: &podInfoOnMount,
+			VolumeLifecycleModes: []storage.VolumeLifecycleMode{
+				storage.VolumeLifecycleEphemeral,
+			},
+		},
+	}
+	var resultEmpty []storage.VolumeLifecycleMode
+	resultPersistent := []storage.VolumeLifecycleMode{storage.VolumeLifecyclePersistent}
+	resultEphemeral := []storage.VolumeLifecycleMode{storage.VolumeLifecycleEphemeral}
+
+	tests := []struct {
+		name                      string
+		old, update               *storage.CSIDriver
+		withInline, withoutInline []storage.VolumeLifecycleMode
+	}{
+		{
+			name:          "before: no mode, update: no mode",
+			old:           driverWithoutModes,
+			update:        driverWithoutModes,
+			withInline:    resultEmpty,
+			withoutInline: resultEmpty,
+		},
+		{
+			name:          "before: no mode, update: persistent",
+			old:           driverWithoutModes,
+			update:        driverWithPersistent,
+			withInline:    resultPersistent,
+			withoutInline: resultEmpty,
+		},
+		{
+			name:          "before: persistent, update: ephemeral",
+			old:           driverWithPersistent,
+			update:        driverWithEphemeral,
+			withInline:    resultEphemeral,
+			withoutInline: resultEphemeral,
+		},
+		{
+			name:          "before: persistent, update: no mode",
+			old:           driverWithPersistent,
+			update:        driverWithoutModes,
+			withInline:    resultEmpty,
+			withoutInline: resultEmpty,
+		},
+	}
+
+	runAll := func(t *testing.T, withInline bool) {
+		for _, test := range tests {
+			t.Run(test.name, func(t *testing.T) {
+				defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIInlineVolume, withInline)()
+
+				csiDriver := test.update.DeepCopy()
+				Strategy.PrepareForUpdate(ctx, csiDriver, test.old)
+				if withInline {
+					require.Equal(t, csiDriver.Spec.VolumeLifecycleModes, test.withInline)
+				} else {
+					require.Equal(t, csiDriver.Spec.VolumeLifecycleModes, test.withoutInline)
+				}
+			})
+		}
+	}
+
+	t.Run("with inline volumes", func(t *testing.T) {
+		runAll(t, true)
+	})
+	t.Run("without inline volumes", func(t *testing.T) {
+		runAll(t, true)
```

Comment:
should be `false`?

Reply:
Indeed. I'll fix that in the next PR or if when there is another reason to update this PR.

Reply:
Fixed.

I had to revise this PR because the CSI mock driver info yaml file wasn't part of it (I incorrectly added it in a later commit that is going to be part of the future testing PR).

@msau42, @jsafrane: please re-add your lgtm.